src/azure/blob.ts

```diff
@@ -43,10 +51,12 @@ export const fromBlobStorage =
       ),
       E.chainW((ctx) =>
         sequenceS(E.Apply)({
-          metadata: validate(
-            metadataSchema,
-            "Unable to validate the metadata schema"
-          )(ctx.bindingData.metadata),
+          metadata: pipe(
+            metadataSchema.decode(ctx.bindingData.metadata),
```

Comment:
it looks like we are discarding the error details (decode result). this would make debugging harder. what about a `details` field into the custom `Error` class?

Reply:
Yes we can add that (I already implemented a custom validation error in the main project, with error details, you will see it in the next PR on `pagopa/io-sign`).

But I don't like the `io-ts` default reporter output, I find it pretty useless for server-to-server communications and, since these errors are not meant to be exposed via APIs (because they can leak critical internal details) I think that the current level of detail (you can discriminate if is a `BlobMetadataParseError`) it's adequate for now.

For validations on user inputs the context is different, so in that case I included the "decode details" into the custom errors (and API error as well)

Reply:
yes but you cannot neither log it in this way (it would be very useful to log decoding errors on bindings)